docs/toolkits/openstack-operations-maintenance.md

```diff
@@ -1,47 +1,181 @@
+openstack role remove --user <user name> --project <project name> <role name>
+```
 
-Checks for a single node database:
+Oftentimes, different projects or users will have different needs of the OpenStack cloud.  This can lead to the default quotas not allowing them to do something that fits their workflow.  For example, perhaps they need more than the default quota of volumes.  These can be adjusted with the CLI by running:
+``` bash
+openstack quota set --volumes <new volume quota> <project name>
+```
+Note that quotas are set per project and not per user.
+
+
+
+### Maintenance
+
+---
+
+#### NTP
+	- What NTP Errors look like
+		The biggest offender here is going to be any clock drift errors in the logs and odd behavior by computes on some isolated VMs.
+	- How to fix
+		Usually this is either two issues.  First, the time daemon running on your node may simply need a restart.  The second is that it could simply be a misconfiguration on said time daemon.
+
+#### Rabbit MQ
+
+	- Checks:
+		To check the status of your RabbitMQ cluster, run the following on one of the rabbitmq nodes:
+``` bash
+rabbitmqctl cluster_status
+```
+Some things to look for here, are any errors or alarms under the [Alarms] section.
+
+	- What rabbitMQ errors look like
+		rabbitMQ errors can be tough to diagnose.  All the openstack services send messages through rabbitMQ, so there is a broad range of errors that an administrator can come across while trying to determine the cause.
+	- How to fix
+		The simplest fix is usually going to be a dead rabbitMQ service on one of the nodes.  Usually a restart can fix these.
+
+#### Mysql
+
+	- Checks for a single node database:
 SSH to the MYSQL node and log into the database with:
 `mysql`
 Also, you can check the logs to make sure there are no discrepancies.
 
-Checks for a multiple node cluster:
+	- Checks for a multiple node cluster:
 SSH to the MYSQL node and log into the database with:
 `mysql`
 and run
 ```
 SHOW GLOBAL STATUS LIKE 'wsrep_%';
 ```
-
 And verify the cluster count matches the number of nodes in your database cluster.
 
-Nova
+	- What MYSQL errors look like
+		If MYSQL is causing issues, you'll usually get a 503 error when attempting to run commands from the CLI.
+	- How to fix
+		This could be as simple as a dead mysql daemon on one of your nodes.  You can diagnose this by running through the checks above.  
+	- Database Discrepancies
+		Throughout the life of your cloud, you may come into issues where the database and openstack reflect different information.  This could be a VM that shows a host on one machine but it's actually on another, a volume that is stuck in a `reserved` state but won't change back to active/available, etc.  To fix this, we need to log into the MYSQL cluster by running `mysql` and updating the discrepancy with `update ITEM set VALUE=DESIRED_VALUE  where id='UUID';`.  You can verify which specific item(s) you'll adjust by adjusting the formula to, `show ITEM where id='UUID';`.
```

Comment:
```suggestion
	- What MySQL errors look like
		If MySQL is causing issues, you'll usually get a 503 error when attempting to run commands from the CLI.
	- How to fix
		This could be as simple as a dead `mysql` daemon on one of your nodes.  You can diagnose this by running through the checks above.  
	- Database Discrepancies
		Throughout the life of your cloud, you may come into issues where the database and OpenStack reflect different information.  This could be a VM that shows a host on one machine but it's actually on another, a volume that is stuck in a `reserved` state but won't change back to active/available, etc.  To fix this, we need to log into the MySQL cluster by running `mysql` and updating the discrepancy with `update ITEM set VALUE=DESIRED_VALUE  where id='UUID';`.  You can verify which specific item(s) you'll adjust by adjusting the formula to, `show ITEM where id='UUID';`.
```

capitalization